alembic_migration/versions/d4d360ef69bc_update_public_transportation_types.py

```diff
@@ -0,0 +1,165 @@
+        sa.Column('public_transportation_types_str', sa.String),
+        schema='guidebook')
+    # For waypoints having 'cable_car' in public_transportation_types:
+    # * set the lift_access flag to true
+    # * remove 'cable_car' from public_transportation_types
+    op.execute(
+        waypoints.update(). \
+        where(and_(
+            waypoints.c.public_transportation_types != None,
+            any_(waypoints.c.public_transportation_types)==op.inline_literal('cable_car')
+        )). \
+        values({
+            'lift_access': True,
+            'public_transportation_types': func.array_remove(
+                waypoints.c.public_transportation_types, 'cable_car')
+        })
+    )
+    # Then fill the new public_transportation_types_ col with values
+    # from public_transportation_types. The intermediary string conversion
+    # is needed because it is not possible to directly cast from one array
+    # type to another.
+    op.execute(
+        waypoints.update(). \
+        where(waypoints.c.public_transportation_types != None). \
+        values({
+            'public_transportation_types_str': func.array_to_string(
+                waypoints.c.public_transportation_types, ',')
+        })
+    )
+    op.execute(
+        waypoints.update(). \
+        where(waypoints.c.public_transportation_types != None). \
+        values({
+            'public_transportation_types_': cast(func.string_to_array(
+                waypoints.c.public_transportation_types_str, ','), ArrayOfEnum(new_type))
+        })
+    )
+
+    # Same for archives data
+    archives = Table(
+        'waypoints_archives',
+        MetaData(),
+        sa.Column('lift_access', sa.Boolean),
+        sa.Column('public_transportation_types', ArrayOfEnum(old_type)),
+        sa.Column('public_transportation_types_', ArrayOfEnum(new_type)),
+        sa.Column('public_transportation_types_str', sa.String),
+        schema='guidebook')
+    op.execute(
+        archives.update(). \
+        where(and_(
+            archives.c.public_transportation_types != None,
+            any_(archives.c.public_transportation_types)==op.inline_literal('cable_car')
+        )). \
+        values({
+            'lift_access': True,
+            'public_transportation_types': func.array_remove(
+                archives.c.public_transportation_types, 'cable_car')
+        })
+    )
+    op.execute(
+        archives.update(). \
+        where(archives.c.public_transportation_types != None). \
+        values({
+            'public_transportation_types_str': func.array_to_string(
+                archives.c.public_transportation_types, ',')
+        })
+    )
+    op.execute(
+        archives.update(). \
+        where(archives.c.public_transportation_types != None). \
+        values({
+            'public_transportation_types_': cast(func.string_to_array(
+                archives.c.public_transportation_types_str, ','), ArrayOfEnum(new_type))
+        })
+    )
```

Comment:
Perhaps the 3 update queries above could be replaced by only 2 like this:
```
    op.execute(
        waypoints.update(). \
        where(waypoints.c.public_transportation_types != None). \
        values({
            'public_transportation_types_str': func.array_to_string(
                func.array_remove(waypoints.c.public_transportation_types, 'cable_car'), ',')
        })
    )
    op.execute(
        waypoints.update(). \
        where(waypoints.c.public_transportation_types != None). \
        values({
            'public_transportation_types_': cast(func.string_to_array(
                waypoints.c.public_transportation_types_str, ','), ArrayOfEnum(new_type))
        })
    )
```
=> we try to remove ``cable_car`` from all entries without testing first if ``waypoints.c.public_transportation_types`` contains it and directly make the ``array_to_string``.
No clue what is more efficient. Anyway the tool is run only once and concerns only ~2000 entries, so I don't think it's really important.